### ISSUE TITLE
Not able to install alibi in conda environment in Linux: Illegal instruction (core dumped)

### ISSUE THREAD

**Issue by sorenwacker** [SeldonIO/alibi#526]
Hi, 

am not able to install and import alibi so far in a conda environment on a Linux system:

> Linux name 4.18.0-305.10.2.el8_4.x86_64 #1 SMP Tue Jul 20 17:25:16 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

I can run `pip install alibi`,

> Successfully installed MarkupSafe-2.0.1 Pillow-8.4.0 PyWavelets-1.2.0 absl-py-0.15.0 alibi-0.6.1 astunparse-1.6.3 attrs-21.2.0 blis-0.7.5 cached-property-1.5.2 cachetools-4.2.4 catalogue-2.0.6 charset-normalizer-2.0.7 clang-5.0 click-8.0.3 cymem-2.0.6 dill-0.3.4 filelock-3.3.2 flatbuffers-1.12 gast-0.4.0 google-auth-1.35.0 google-auth-oauthlib-0.4.6 google-pasta-0.2.0 grpcio-1.41.1 h5py-3.1.0 huggingface-hub-0.1.2 idna-3.3 imageio-2.10.3 importlib-metadata-4.8.2 jinja2-3.0.3 keras-2.6.0 keras-preprocessing-1.1.2 langcodes-3.2.1 markdown-3.3.4 murmurhash-1.0.6 networkx-2.6.3 oauthlib-3.1.1 opt-einsum-3.3.0 packaging-21.2 pathy-0.6.1 preshed-3.0.6 protobuf-3.19.1 pyasn1-0.4.8 pyasn1-modules-0.2.8 pydantic-1.8.2 pyparsing-2.4.7 pyyaml-6.0 regex-2021.11.10 requests-2.26.0 requests-oauthlib-1.3.0 rsa-4.7.2 sacremoses-0.0.46 scikit-image-0.18.3 scikit-learn-0.24.2 six-1.15.0 smart-open-5.2.1 spacy-3.2.0 spacy-legacy-3.0.8 spacy-loggers-1.0.1 spacy-lookups-data-1.0.3 srsly-2.4.2 tensorboard-2.6.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.0 tensorflow-2.6.2 tensorflow-estimator-2.6.0 termcolor-1.1.0 thinc-8.0.13 tifffile-2021.11.2 tokenizers-0.10.3 tqdm-4.62.3 transformers-4.12.3 typer-0.4.0 typing-extensions-3.7.4.3 urllib3-1.26.7 wasabi-0.8.2 werkzeug-2.0.2 wrapt-1.12.1 zipp-3.6.0

 but when I try to import it I get:

    python -c 'import alibi'
    > Illegal instruction (core dumped)

I tried different Python versions 3.9 and 3.7 and different scipy versions 1.4.1, 1.5.2, 1.7.2. 

Nothing helps.

**Comment by jklaise** [comment]
This is very likely either a `tensorflow` or potentially a `numpy` version, can you try to isolate the problem by importing just one of these?

**Comment by jklaise** [comment]
Also see https://github.com/tensorflow/tensorflow/issues/52897

**Comment by sorenwacker** [comment]
Confirmed. It comes from tensorflow:
```
python -c 'import tensorflow'
Illegal instruction (core dumped)
```

**Comment by ascillitoe** [comment]
@jklaise @mauicv since there isn't much we can do about this on our end, shall we close this? Hopefully, modularisation will help with this in the future...

**Comment by jklaise** [comment]
Yes, this is purely an issue with `tensorflow` and the environment it's being installed in.
